docs/source/extensions/datamodules.rst

```diff
@@ -292,76 +299,68 @@ Returns a special dataloader for inference. This is the dataloader that the Trai
 
     class MNISTDataModule(pl.LightningDataModule):
         def predict_dataloader(self):
-            return DataLoader(self.mnist_test, batch_size=64)
+            return DataLoader(self.mnist_predict, batch_size=64)
 
 
 transfer_batch_to_device
 ^^^^^^^^^^^^^^^^^^^^^^^^
-Override to define how you want to move an arbitrary batch to a device.
-To check the current state of execution of this hook you can use ``self.trainer.training/testing/validating/predicting/sanity_checking``
-so that you can add different logic as per your requirement.
-
-.. testcode::
-
-    class MNISTDataModule(LightningDataModule):
-        def transfer_batch_to_device(self, batch, device, dataloader_idx):
-            x = batch["x"]
-            x = CustomDataWrapper(x)
-            batch["x"] = x.to(device)
-            return batch
-
-
-.. note:: This hook only runs on single GPU training and DDP (no data-parallel).
 
+.. automethod:: pytorch_lightning.core.datamodule.LightningDataModule.transfer_batch_to_device
+    :noindex:
 
 on_before_batch_transfer
 ^^^^^^^^^^^^^^^^^^^^^^^^
-Override to alter or apply augmentations to your batch before it is transferred to the device.
-To check the current state of execution of this hook you can use ``self.trainer.training/testing/validating/predicting/sanity_checking``
-so that you can add different logic as per your requirement.
 
-.. testcode::
+.. automethod:: pytorch_lightning.core.datamodule.LightningDataModule.on_before_batch_transfer
+    :noindex:
 
-    class MNISTDataModule(LightningDataModule):
-        def on_before_batch_transfer(self, batch, dataloader_idx):
-            batch["x"] = transforms(batch["x"])
-            return batch
+on_after_batch_transfer
+^^^^^^^^^^^^^^^^^^^^^^^
 
+.. automethod:: pytorch_lightning.core.datamodule.LightningDataModule.on_after_batch_transfer
+    :noindex:
 
-.. note:: This hook only runs on single GPU training and DDP (no data-parallel).
+on_load_checkpoint
+~~~~~~~~~~~~~~~~~~
 
```

Comment:
from here on, the title levels are inconsistent, causing the index to be confused:

![image](https://user-images.githubusercontent.com/5495193/143036244-2fd9d085-7fe3-4e3c-8549-c473eb988c3f.png)

Not sure why.

Reply:
fixed!